controllers/secretgenerator/secretgenerator_controller.go

```diff
@@ -109,15 +104,22 @@ func (r *SecretGeneratorReconciler) Reconcile(ctx context.Context, request ctrl.
 	log := r.Log
 	foundSecret := &corev1.Secret{}
 	err := r.Client.Get(ctx, request.NamespacedName, foundSecret)
+
+	// deletion case
 	if err != nil {
-		if k8serr.IsNotFound(err) {
-			// If Secret is deleted, delete OAuthClient if exists
-			err = r.deleteOAuthClient(ctx, request.Name)
+		if k8serr.IsNotFound(err) || foundSecret.GetDeletionTimestamp() != nil {
+			r.Log.Info("Reconciling Secret on deletion.", "Request.Name", request.Name)
+			// delete OAuthClient if exists
+			if err = r.deleteOAuthClient(ctx, request.Name); err != nil {
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{}, nil
 		}
-
 		return ctrl.Result{}, err
 	}
 
+	r.Log.Info("Reconciling Secret on creation.", "Request.Name", request.Name)
```

Comment:
Shouldn't this be
```
log := r.Log
log.Info(...)
```
to align with https://github.com/opendatahub-io/opendatahub-operator/pull/1272?

Reply:
thanks , updated